azurelinuxagent/ga/exthandlers.py

```diff
@@ -312,8 +312,16 @@ def handle_ext_handler(self, ext_handler, etag):
 
         try:
             state = ext_handler.properties.state
+            max_retries = 5
+            retry_count = 0
             while ext_handler_i.decide_version(target_state=state) is None:
-                continue
+                if retry_count >= max_retries:
+                    err_msg = "Unable to find manifest for extension {0}".format(ext_handler_i.ext_handler.name)
+                    ext_handler_i.set_handler_status(message=ustr(err_msg), code=-1)
+                    ext_handler_i.report_event(message=ustr(err_msg), is_success=False)
+                    return
+                time.sleep(2**retry_count * 10)
+                retry_count += 1
             if not ext_handler_i.is_upgrade and self.last_etag == etag:
```

Comment:
If some retries were required but we eventually succeeded, I think we should still emit a telemetry event that summarizes the number of retries and the total delay before success. I can see wanting to put an alert on this, since "having to retry" might be an early warning sign of an incipient failure we could head off. No need to set handler status; just emit the telemetry event.

Reply:
I agree we should have telemetry around transient failures.